### ISSUE TITLE
"command '_workbench.downloadResource' failed" when connecting to a remote host via ssh

### ISSUE THREAD

**Issue by massimo-pellagatti** [microsoft/vscode-remote-release#4743]
- VS Code Version: 1.55.0-insider (user setup)
- OS Version: Windows 10 Enterprise (Version 18039)

Steps to Reproduce:

1. Click on the *remote explorer* icon in the taskbar
2. Select a remote ssh target
3. click on the icon *Open SSH host in new window*

Does this issue occur when all extensions are disabled?: Yes

Up to yesterday I was able to connect to the host, but since today I'm not able anymore. I tried connecting to the host via the command *ssh user@hostname* and I had no problems.  
A colleague of mine is able to connect to the host via vscode. This is its vscode help->about: 

Version: 1.55.0-insider (user setup)
Commit: 84fe402d655e029eb1a5c04e675bf64788fa7fcf
Date: 2021-03-15T05:14:29.836Z
Electron: 11.3.0
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: Windows_NT x64 10.0.17134

In the attached files you can find 
- the screenshot of the popup
- my ssh config file
- the logs
- the help->about of my VSCode
- the extension I installed

EDIT: It seems I'm not able to attach the files, I select them, but then nothing happens, I leave a [link](https://gofile.io/d/i8AxNj) to access them

**Comment by chrmarti** [comment]
@massimo-pellagatti Could you start VS Code with `--log-net-log=netlog.json`, retry to connect to the SSH remote and then post the created netlog.json here?

**Comment by massimo-pellagatti** [comment]
Hello,

the problem, as it came, seems to have gone away: today I was able to connect to my desired host with no issue. Also another colleague of mine, who had the same problem, today can access the remote server without issues.
The commit id of my VSCode insiders version is the same, so maybe it wasn't a VSCode problem (even though yesterday I could connect to the server with the *ssh* command).

However, I share the files you asked for, just in case they might be helpful for future problems or in case the problem I had yesterday comes back.  
At [this](https://gofile.io/d/nvch9o) link you can see (I still don't understand why I can't attach files in the way I'm supposed to, sorry for spamming links to external websites) :
- the *netlog.json* file
- the output of `Help > Toggle Developer Tools > Console`
- the logs of the access happened without errors.

**Comment by firegate22** [comment]
Some logs - same issue - but no solution (even after days).  I tried to delete local workspaceStorage and stopped/deleted the vscode-insiders server via vscode and by deleting it via ssh/terminal on the remote machine.

Insiders 1.58 with problems... and stable 1.57.1 one works. Remote-SSH both 0.65.7.

Logs:
`[09:02:59.703] Log Level: 2
[09:02:59.714] remote-ssh@0.65.7
[09:02:59.715] win32 x64
[09:02:59.754] SSH Resolver called for "ssh-remote+<<ssh url>>", attempt 1
[09:02:59.755] "remote.SSH.useLocalServer": false
[09:02:59.755] "remote.SSH.showLoginTerminal": false
[09:02:59.756] "remote.SSH.remotePlatform": {"<<ssh url>>":"linux"}
[09:02:59.756] "remote.SSH.path": undefined
[09:02:59.756] "remote.SSH.configFile": undefined
[09:02:59.756] "remote.SSH.useFlock": true
[09:02:59.756] "remote.SSH.lockfilesInTmp": false
[09:02:59.756] "remote.SSH.localServerDownload": always
[09:02:59.757] "remote.SSH.remoteServerListenOnSocket": false
[09:02:59.757] "remote.SSH.showLoginTerminal": false
[09:02:59.757] "remote.SSH.defaultExtensions": []
[09:02:59.757] "remote.SSH.loglevel": 2
[09:02:59.757] SSH Resolver called for host: <<ssh url>>
[09:02:59.757] Setting up SSH remote "<<ssh url>>"
[09:02:59.764] Using commit id "a81fff00c9dab105800118fcf8b044cd84620419" and quality "insider" for server
[09:02:59.768] Install and start server if needed
[09:02:59.772] Checking ssh with "ssh -V"
[09:02:59.906] > OpenSSH_for_Windows_8.1p1, 
[09:02:59.906] > LibreSSL 3.0.2

[09:02:59.912] Running script with connection command: ssh -T -D 49222 "<<ssh url>>" bash
[09:02:59.915] Terminal shell path: C:\WINDOWS\System32\cmd.exe
[09:03:00.350] > �]0;C:\WINDOWS\System32\cmd.exe�
[09:03:00.350] Got some output, clearing connection timeout
[09:03:01.233] > 80575b888811: running
[09:03:01.266] > Acquiring lock on /home/<<my username>>/.vscode-server-insiders/bin/a81fff00c9dab105
> 800118fcf8b044cd84620419/vscode-remote-lock.<<my username>>.a81fff00c9dab105800118fc
> f8b044cd84620419
[09:03:01.280] > Installing to /home/<<my username>>/.vscode-server-insiders/bin/a81fff00c9dab1058001
> 18fcf8b044cd84620419...
> Trigger local server download
> 80575b888811:trigger_server_download
> platform==linux==
> vscodeArch==x64==
> destFolder==/home/<<my username>>/.vscode-server-insiders/bin/a81fff00c9dab105800118
> fcf8b044cd84620419==
> 80575b888811:trigger_server_download_end
> Waiting for client to transfer server archive...
> Waiting for /home/<<my username>>/.vscode-server-insiders/bin/a81fff00c9dab105800118
> fcf8b044cd84620419/vscode-scp-done.flag and vscode-server.tar.gz to exist       
> 
[09:03:01.281] Got request to download on client for {"platform":"linux","arch":"x64","destFolder":"/home/<<my username>>/.vscode-server-insiders/bin/a81fff00c9dab105800118fcf8b044cd84620419"}
[09:03:01.281] Downloading VS Code server locally...
[09:03:01.486] Resolver error: Error: Running the contributed command: '_workbench.downloadResource' failed.
	at I._executeContributedCommand (c:\Users\<<my username>>\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:84:108993)
	at processTicksAndRejections (internal/process/task_queues.js:93:5)
	at async c:\Users\<<my username>>\.vscode-insiders\extensions\ms-vscode-remote.remote-ssh-0.65.7\out\extension.js:1:139384
	at async Object.t.downloadAndCopyToHost (c:\Users\<<my username>>\.vscode-insiders\extensions\ms-vscode-remote.remote-ssh-0.65.7\out\extension.js:1:139299)
	at async c:\Users\<<my username>>\.vscode-insiders\extensions\ms-vscode-remote.remote-ssh-0.65.7\out\extension.js:1:414617
	at async Object.t.tryInstall (c:\Users\<<my username>>\.vscode-insiders\extensions\ms-vscode-remote.remote-ssh-0.65.7\out\extension.js:1:414829)
	at async c:\Users\<<my username>>\.vscode-insiders\extensions\ms-vscode-remote.remote-ssh-0.65.7\out\extension.js:1:294918
	at async Object.t.withShowDetailsEvent (c:\Users\<<my username>>\.vscode-insiders\extensions\ms-vscode-remote.remote-ssh-0.65.7\out\extension.js:1:406463)
	at async Object.t.resolve (c:\Users\<<my username>>\.vscode-insiders\extensions\ms-vscode-remote.remote-ssh-0.65.7\out\extension.js:1:295994)
	at async c:\Users\<<my username>>\.vscode-insiders\extensions\ms-vscode-remote.remote-ssh-0.65.7\out\extension.js:127:110656
[09:03:01.489] ------`

**Comment by roblourens** [comment]
CORS issue on the update server 
![image](https://user-images.githubusercontent.com/323878/123691075-06c86100-d80a-11eb-8d8f-f7d098ecbdb3.png)

**Comment by joaomoreno** [comment]
Why is this request being made from the browser? 🤔 Is this web?

**Comment by joaomoreno** [comment]
Do we fallback to `_workbench.downloadResource` if the remote can't download the agent? I guess that's it. Yeah this might be an oversight from adopting `vscode-file` cc @bpasero.

**Comment by leifwalsh** [comment]
In my case, our Windows (client) environment is much easier to configure correctly and transparently than the Linux hosts we're remoting to, so for a good user experience we prefer `"remote.SSH.localServerDownload": "always"`, which worked on Insiders until fairly recently. I think we're experiencing the same problem, and I'm trying to get approval to upload you a netlog of it. But if you think you're on the right track with this CORS and `vscode-file://` thing, hopefully you know what you need already.

**Comment by bpasero** [comment]
With the change to using `vscode-file` as scheme, CORS policies apply to VSCode desktop. This was discussed in https://github.com/microsoft/vscode-internalbacklog/issues/1824. If there is another service we forgot, that service will need to be aware of the new origin.

**Comment by frankli0324** [comment]
this seems to have affected stable builds
btw, 404 for the internalbacklog repo
![image](https://user-images.githubusercontent.com/20221896/124971425-bc41b280-e05b-11eb-8c9b-c34324cbc640.png)

**Comment by jopbrown** [comment]
stable version 1.58.0
```
Access to XMLHttpRequest at 'https://vscode.cdn.azure.cn/stable/2d23c42a936db1c7b3b06f918cde29561cc47cd6/vscode-server-linux-x64.tar.gz' (redirected from 'https://update.code.visualstudio.com/commit%3A2d23c42a936db1c7b3b06f918cde29561cc47cd6/server-linux-x64/stable') from origin 'vscode-file://vscode-app' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
```
@joaomoreno 
should reopen the issue?

**Comment by GeniusPatrick** [comment]
Same issue here.
After updating vscode to stable version 1.58.0
remote-ssh version 0.65.7

**Comment by liuyatao** [comment]
I meet the same problem

**Comment by ippfcox** [comment]
Same here

I downloaded right vscode-server.tar.gz(https://vscode.cdn.azure.cn/stable/2d23c42a936db1c7b3b06f918cde29561cc47cd6/vscode-server-linux-x64.tar.gz) via wget manually to the path ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6
Then touch  vscode-scp-done.flag when connecting 

server can be installed successfully...

this maybe a temporary solution if you are busy working with VSCode

**Comment by joaomoreno** [comment]
I assume you're all in China?

@roblourens Is there any specific reason why so many people are now hitting the alternative route of downloading the server from the client? I thought the default route was via the remote machine itself, using your scripts.

**Comment by frankli0324** [comment]
> I thought the default route was via the remote machine itself, using your scripts.

I set `local download` to `always` because I have several machines behind firewalls without internet access.

**Comment by walterzhaoJR** [comment]
> Same here
> 
> I downloaded right vscode-server.tar.gz(https://vscode.cdn.azure.cn/stable/2d23c42a936db1c7b3b06f918cde29561cc47cd6/vscode-server-linux-x64.tar.gz) via wget manually to the path ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6
> Then touch vscode-scp-done.flag when connecting
> 
> server can be installed successfully...
> 
> this maybe a temporary solution if you are busy working with VSCode

This is useful, great

**Comment by ippfcox** [comment]
> I assume you're all in China?
> 
> @roblourens Is there any specific reason why so many people are now hitting the alternative route of downloading the server from the client? I thought the default route was via the remote machine itself, using your scripts.

Emm yes, but this is the first time I meet this issue, and not all my remote machines fail updating. One succeeded since the other failed for me.

**Comment by joaomoreno** [comment]
Reopening, even though this is a separate issue as the initial one. The China CDN needs to open up its CORS rules.

**Comment by goopand** [comment]
> Same here
> 
> I downloaded right vscode-server.tar.gz(https://vscode.cdn.azure.cn/stable/2d23c42a936db1c7b3b06f918cde29561cc47cd6/vscode-server-linux-x64.tar.gz) via wget manually to the path ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6
> Then touch vscode-scp-done.flag when connecting
> 
> server can be installed successfully...
> 
> this maybe a temporary solution if you are busy working with VSCode

i followed this solution and it worked. btw, detailed procedures are as below:
1) upload the right vscode-server.tar.gz(https://vscode.cdn.azure.cn/stable/2d23c42a936db1c7b3b06f918cde29561cc47cd6/vscode-server-linux-x64.tar.gz) to ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6 
2) cd ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6 ; tar -zxf vscode-server-linux-x64.tar.gz ; mv vscode-server-linux-x64 vscode-server
3) touch vscode-scp-done.flag
4) click 'Retry' in VSCode window

**Comment by qihangxu903** [comment]
> > Same here
> > I downloaded right vscode-server.tar.gz(https://vscode.cdn.azure.cn/stable/2d23c42a936db1c7b3b06f918cde29561cc47cd6/vscode-server-linux-x64.tar.gz) via wget manually to the path ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6
> > Then touch vscode-scp-done.flag when connecting
> > server can be installed successfully...
> > this maybe a temporary solution if you are busy working with VSCode
> 
> i followed this solution and it worked. btw, detailed procedures are as below:
> 
> 1. upload the right vscode-server.tar.gz(https://vscode.cdn.azure.cn/stable/2d23c42a936db1c7b3b06f918cde29561cc47cd6/vscode-server-linux-x64.tar.gz) to ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6
> 2. cd ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6 ; tar -zxf vscode-server-linux-x64.tar.gz ; mv vscode-server-linux-x64 vscode-server
> 3. touch vscode-scp-done.flag
> 4. click 'Retry' in VSCode window

Thanks for the detailed steps, work like a charm.

**Comment by SoloGao** [comment]
> > Same here
> > I downloaded right vscode-server.tar.gz(https://vscode.cdn.azure.cn/stable/2d23c42a936db1c7b3b06f918cde29561cc47cd6/vscode-server-linux-x64.tar.gz) via wget manually to the path ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6
> > Then touch vscode-scp-done.flag when connecting
> > server can be installed successfully...
> > this maybe a temporary solution if you are busy working with VSCode
> 
> i followed this solution and it worked. btw, detailed procedures are as below:
> 
> 1. upload the right vscode-server.tar.gz(https://vscode.cdn.azure.cn/stable/2d23c42a936db1c7b3b06f918cde29561cc47cd6/vscode-server-linux-x64.tar.gz) to ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6
> 2. cd ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6 ; tar -zxf vscode-server-linux-x64.tar.gz ; mv vscode-server-linux-x64 vscode-server
> 3. touch vscode-scp-done.flag
> 4. click 'Retry' in VSCode window

Great! That saves my day.
Besides, you may need to mv vscode-server-linux-x64.tar.gz vscode-server.tar.gz and change Remote - SSH -> Local Server Download to "auto" to disable local download trial.

**Comment by jinzhaochaliang** [comment]
> > Same here
> > I downloaded right vscode-server.tar.gz(https://vscode.cdn.azure.cn/stable/2d23c42a936db1c7b3b06f918cde29561cc47cd6/vscode-server-linux-x64.tar.gz) via wget manually to the path ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6
> > Then touch vscode-scp-done.flag when connecting
> > server can be installed successfully...
> > this maybe a temporary solution if you are busy working with VSCode
> 
> i followed this solution and it worked. btw, detailed procedures are as below:
> 
> 1. upload the right vscode-server.tar.gz(https://vscode.cdn.azure.cn/stable/2d23c42a936db1c7b3b06f918cde29561cc47cd6/vscode-server-linux-x64.tar.gz) to ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6
> 2. cd ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6 ; tar -zxf vscode-server-linux-x64.tar.gz ; mv vscode-server-linux-x64 vscode-server
> 3. touch vscode-scp-done.flag
> 4. click 'Retry' in VSCode window

Thank you for the detailed steps.

**Comment by krzikalla** [comment]
Same here (location: Germany). Happens on all my remote SSH nodes, got one fixed with the steps above (thank you). Worked like a charm before the weekend. 
In addition it has started to annoy me about a german language pack - not sure, if related.

**Comment by LarryDong** [comment]
Hi I met the same problem after I update to vscode-1.58 with ssh 0.65.7
I notice that the ssh on server try to update something but failed all the time, I then realized that my server is not connected to the Internet.
So after connecting the Internet,  it's like a miracle, it disappeared. 
Hope it helps a little.

**Comment by joaomoreno** [comment]
Moving the issue to https://github.com/microsoft/vscode/issues/128492.

**Comment by yintao-he** [comment]
> Hi I met the same problem after I update to vscode-1.58 with ssh 0.65.7
> I notice that the ssh on server try to update something but failed all the time, I then realized that my server is not connected to the Internet.
> So after connecting the Internet, like a miracle, this issue disappeared.
> Hope it helps a little.

The same problem has been fixed after my server connecting the Internet.
Thx for your help : )

**Comment by linkjam** [comment]
> Hi I met the same problem after I update to vscode-1.58 with ssh 0.65.7
> I notice that the ssh on server try to update something but failed all the time, I then realized that my server is not connected to the Internet.
> So after connecting the Internet, like a miracle, this issue disappeared.
> Hope it helps a little.

but many server have no Internet access, they can work under LAN only😔

**Comment by DCupRobot** [comment]
Just close the auto update and uninstall vs code 1.58.0 and install 1.57.x or other version

**Comment by freshweek** [comment]
I have same problem, and I solve it after setting proxy for vscode.
![image](https://user-images.githubusercontent.com/43909206/125904383-02882888-470d-422a-8d65-19056650be5b.png)

**Comment by Tiger1994** [comment]
> > Same here
> > I downloaded right vscode-server.tar.gz(https://vscode.cdn.azure.cn/stable/2d23c42a936db1c7b3b06f918cde29561cc47cd6/vscode-server-linux-x64.tar.gz) via wget manually to the path ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6
> > Then touch vscode-scp-done.flag when connecting
> > server can be installed successfully...
> > this maybe a temporary solution if you are busy working with VSCode
> 
> i followed this solution and it worked. btw, detailed procedures are as below:
> 
> 1. upload the right vscode-server.tar.gz(https://vscode.cdn.azure.cn/stable/2d23c42a936db1c7b3b06f918cde29561cc47cd6/vscode-server-linux-x64.tar.gz) to ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6
> 2. cd ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6 ; tar -zxf vscode-server-linux-x64.tar.gz ; mv vscode-server-linux-x64 vscode-server
> 3. touch vscode-scp-done.flag
> 4. click 'Retry' in VSCode window

thanks very much, I follow your steps, and get the follow error:

> Failed to connect to the remote extension host server (Error: Connection error: Version mismatch, client refused.

and I clean the old pid files, like this:
![image](https://user-images.githubusercontent.com/25787227/125905757-ccbee206-e6a2-4036-b28b-ffadda9b22fa.png)

then the problem solved.

**Comment by ippfcox** [comment]
> > Same here
> > I downloaded right vscode-server.tar.gz(https://vscode.cdn.azure.cn/stable/2d23c42a936db1c7b3b06f918cde29561cc47cd6/vscode-server-linux-x64.tar.gz) via wget manually to the path ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6
> > Then touch vscode-scp-done.flag when connecting
> > server can be installed successfully...
> > this maybe a temporary solution if you are busy working with VSCode
> 
> i followed this solution and it worked. btw, detailed procedures are as below:
> 
> 1. upload the right vscode-server.tar.gz(https://vscode.cdn.azure.cn/stable/2d23c42a936db1c7b3b06f918cde29561cc47cd6/vscode-server-linux-x64.tar.gz) to ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6
> 2. cd ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6 ; tar -zxf vscode-server-linux-x64.tar.gz ; mv vscode-server-linux-x64 vscode-server
> 3. touch vscode-scp-done.flag
> 4. click 'Retry' in VSCode window

This problem(CORS) occurs again for me now🙄, if somebody else meets this too, please change 2d23c42a936db1c7b3b06f918cde29561cc47cd6 to c3f126316369cd610563c75b1b1725e0679adfb3 and follow these steps.

**Comment by erhsh** [comment]
> > > Same here
> > > I downloaded right vscode-server.tar.gz(https://vscode.cdn.azure.cn/stable/2d23c42a936db1c7b3b06f918cde29561cc47cd6/vscode-server-linux-x64.tar.gz) via wget manually to the path ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6
> > > Then touch vscode-scp-done.flag when connecting
> > > server can be installed successfully...
> > > this maybe a temporary solution if you are busy working with VSCode
> > 
> > 
> > i followed this solution and it worked. btw, detailed procedures are as below:
> > 
> > 1. upload the right vscode-server.tar.gz(https://vscode.cdn.azure.cn/stable/2d23c42a936db1c7b3b06f918cde29561cc47cd6/vscode-server-linux-x64.tar.gz) to ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6
> > 2. cd ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6 ; tar -zxf vscode-server-linux-x64.tar.gz ; mv vscode-server-linux-x64 vscode-server
> > 3. touch vscode-scp-done.flag
> > 4. click 'Retry' in VSCode window
> 
> This problem(CORS) occurs again for me now🙄, if somebody else meets this too, please change 2d23c42a936db1c7b3b06f918cde29561cc47cd6 to c3f126316369cd610563c75b1b1725e0679adfb3 and follow these steps.

Same problem!

**Comment by frankli0324** [comment]
> Moving the issue to [microsoft/vscode#128492](https://github.com/microsoft/vscode/issues/128492).

@erhsh @ippfcox please follow the new thread

**Comment by standingbychen** [comment]
> > Same here
> > I downloaded right vscode-server.tar.gz(https://vscode.cdn.azure.cn/stable/2d23c42a936db1c7b3b06f918cde29561cc47cd6/vscode-server-linux-x64.tar.gz) via wget manually to the path ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6
> > Then touch vscode-scp-done.flag when connecting
> > server can be installed successfully...
> > this maybe a temporary solution if you are busy working with VSCode
> 
> i followed this solution and it worked. btw, detailed procedures are as below:
> 
> 1. upload the right vscode-server.tar.gz(https://vscode.cdn.azure.cn/stable/2d23c42a936db1c7b3b06f918cde29561cc47cd6/vscode-server-linux-x64.tar.gz) to ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6
> 2. cd ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6 ; tar -zxf vscode-server-linux-x64.tar.gz ; mv vscode-server-linux-x64 vscode-server
> 3. touch vscode-scp-done.flag
> 4. click 'Retry' in VSCode window

Thank you for your specific solution.
I find that the code '2d23c42a936db1c7b3b06f918cde29561cc47cd6' is dynamic which means it may be different to different people and time.
I downloaded the file and operated in path. But I still can't get remote host.
While I find that there is another empty dir named by code '379476f0e13988d90fab105c5c19e7abc8b1dea8'. I operated in the path and get success.

**Comment by weimengxi** [comment]
> stable version 1.58.0
> 
> ```
> Access to XMLHttpRequest at 'https://vscode.cdn.azure.cn/stable/2d23c42a936db1c7b3b06f918cde29561cc47cd6/vscode-server-linux-x64.tar.gz' (redirected from 'https://update.code.visualstudio.com/commit%3A2d23c42a936db1c7b3b06f918cde29561cc47cd6/server-linux-x64/stable') from origin 'vscode-file://vscode-app' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
> ```
> 
> @joaomoreno
> should reopen the issue?

same with you

**Comment by kayoch1n** [comment]
> > Same here
> > I downloaded right vscode-server.tar.gz(https://vscode.cdn.azure.cn/stable/2d23c42a936db1c7b3b06f918cde29561cc47cd6/vscode-server-linux-x64.tar.gz) via wget manually to the path ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6
> > Then touch vscode-scp-done.flag when connecting
> > server can be installed successfully...
> > this maybe a temporary solution if you are busy working with VSCode
> 
> i followed this solution and it worked. btw, detailed procedures are as below:
> 
> 1. upload the right vscode-server.tar.gz(https://vscode.cdn.azure.cn/stable/2d23c42a936db1c7b3b06f918cde29561cc47cd6/vscode-server-linux-x64.tar.gz) to ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6
> 2. cd ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6 ; tar -zxf vscode-server-linux-x64.tar.gz ; mv vscode-server-linux-x64 vscode-server
> 3. touch vscode-scp-done.flag
> 4. click 'Retry' in VSCode window

For people who suffer from mismatched version

> > > Same here
> > > I downloaded right vscode-server.tar.gz(https://vscode.cdn.azure.cn/stable/2d23c42a936db1c7b3b06f918cde29561cc47cd6/vscode-server-linux-x64.tar.gz) via wget manually to the path ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6
> > > Then touch vscode-scp-done.flag when connecting
> > > server can be installed successfully...
> > > this maybe a temporary solution if you are busy working with VSCode
> > 
> > 
> > i followed this solution and it worked. btw, detailed procedures are as below:
> > 
> > 1. upload the right vscode-server.tar.gz(https://vscode.cdn.azure.cn/stable/2d23c42a936db1c7b3b06f918cde29561cc47cd6/vscode-server-linux-x64.tar.gz) to ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6
> > 2. cd ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6 ; tar -zxf vscode-server-linux-x64.tar.gz ; mv vscode-server-linux-x64 vscode-server
> > 3. touch vscode-scp-done.flag
> > 4. click 'Retry' in VSCode window
> 
> Thank you for your specific solution.
> I find that the code '2d23c42a936db1c7b3b06f918cde29561cc47cd6' is dynamic, which means it may be different in different time or version of something.
> I downloaded the file and operated in path. But I still can't get remote host.
> While, I find that there is another empty dir named by code '379476f0e13988d90fab105c5c19e7abc8b1dea8'. I guess that it is created by vscode, but the vscode failed to download files into it. I download files from url https://vscode.cdn.azure.cn/stable/379476f0e13988d90fab105c5c19e7abc8b1dea8/vscode-server-linux-x64.tar.gz, operate steps above in the path `~/.vscode-server/bin/379476f0e13988d90fab105c5c19e7abc8b1dea8/` and make it.

The code string should match your client version. For Mac users, they can find it by "Code" -> "About Visual Studio Code"

**Comment by westion717** [comment]
Could you guys fix it in ONE shot?

**Comment by planetPocket** [comment]
> > > > Same here
> > > > I downloaded right vscode-server.tar.gz(https://vscode.cdn.azure.cn/stable/2d23c42a936db1c7b3b06f918cde29561cc47cd6/vscode-server-linux-x64.tar.gz) via wget manually to the path ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6
> > > > Then touch vscode-scp-done.flag when connecting
> > > > server can be installed successfully...
> > > > this maybe a temporary solution if you are busy working with VSCode
> > > 
> > > 
> > > i followed this solution and it worked. btw, detailed procedures are as below:
> > > 
> > > 1. upload the right vscode-server.tar.gz(https://vscode.cdn.azure.cn/stable/2d23c42a936db1c7b3b06f918cde29561cc47cd6/vscode-server-linux-x64.tar.gz) to ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6
> > > 2. cd ~/.vscode-server/bin/2d23c42a936db1c7b3b06f918cde29561cc47cd6 ; tar -zxf vscode-server-linux-x64.tar.gz ; mv vscode-server-linux-x64 vscode-server
> > > 3. touch vscode-scp-done.flag
> > > 4. click 'Retry' in VSCode window
> > 
> > 
> > Thank you for your specific solution.
> > I find that the code '2d23c42a936db1c7b3b06f918cde29561cc47cd6' is dynamic, which means it may be different in different time or version of something.
> > I downloaded the file and operated in path. But I still can't get remote host.
> > While, I find that there is another empty dir named by code '379476f0e13988d90fab105c5c19e7abc8b1dea8'. I guess that it is created by vscode, but the vscode failed to download files into it. I download files from url https://vscode.cdn.azure.cn/stable/379476f0e13988d90fab105c5c19e7abc8b1dea8/vscode-server-linux-x64.tar.gz, operate steps above in the path `~/.vscode-server/bin/379476f0e13988d90fab105c5c19e7abc8b1dea8/` and make it.
> 
> The code string should match your client's version(or "commit"). For Mac users, they can find it by "Code" -> "About Visual Studio Code"
> 
> > Version: 1.59.0
> > Commit: 379476f0e13988d90fab105c5c19e7abc8b1dea8
> > Date: 2021-08-04T23:14:40.191Z
> > Electron: 13.1.7
> > Chrome: 91.0.4472.124
> > Node.js: 14.16.0
> > V8: 9.1.269.36-electron.0
> > OS: Darwin x64 19.3.0

After the above operations, I got two errors when I try to connect my linux server(offline):

> Could not fetch remote environment.
> Failed to connect to the remote extension host server (Error: Connection error: Client refused: version mismatch)

vscode version:

> Version: 1.59.0 (user setup)
> Commit: 379476f0e13988d90fab105c5c19e7abc8b1dea8
> OS: Windows_NT x64 10.0.19041
